code/segmentedtickerplant/stplog.q

```diff
@@ -209,7 +210,7 @@ endofperiod:{[currentpd;nextpd;data]
 stpeoperiod:{[currentpd;nextpd;data;rolllogs]
   .lg.o[`stpeoperiod;"passing on endofperiod message to subscribers"];
   .stpps.endp[currentpd;nextpd;data];                      // sends endofperiod message to subscribers
-  currperiod::nextperiod;                                  // increments current period
+  currperiod::nextpd;                                      // increments current period
```

Comment:
is there a reason for this to be passed as an argument rather than use the global? I see on the next line that the global is updated, is it valid for anything _other_ than the global to be passed as `nextpd`?

Suppose what I'm really asking is, should the change instead be to remove the `nextpd` argument from this function and update every call to this function?

Reply:
The log message at the end of this function (line 218) specifically uses the `currentpd` & `nextpd` arguments, which at this stage would have different values to `currperiod` & `nextperiod`.
That said, it occurs to me that a log message for "new values for current and next period" ought to use the freshly updated `currperiod` & `nextperiod` instead - in which case, the `currentpd` & `nextpd` arguments could be dropped from `stpeoperiod[]` without consequence.

Reply:
yep, I agree, that log message is currently wrong